DRAFT.md

```diff
@@ -8,30 +8,141 @@ Working Group: Library, Core, Evolution
+As a language-level abstraction in a low-level language, they have even more problems.
 
 ## What are fibers?
 
-TODO
+Fibres are a building block for N:M threading, allowing a set of thread-like abstractions to be multiplexed onto a single scheduled entity.
+In the general case, they may be cooperatively scheduled (with explicit yielding) or preemptively scheduled in response to timer events.
+The goal for fibers is to be faster for context switching than normal thread switching.
 
 ## What is the current proposal?
 
+The current proposal is for fibers independent of any kind of scheduler.
+Schedulers 
+
 ## Issues with the current proposal
 
-### Dependency on deprecated POSIX features
+### What is a context?
+
+Fibres intended for userspace context switches. The root problem here is defining what a context is.  In the simplest case, it’s the contents of a register file, which includes a stack pointer.  This is not sufficient for most operating systems.  Other state may include:
+
+ - The current signal mask.
+ - Pending signals or asynchronous callbacks.
+ - In-flight asynchronous I/O
+ - And so on.
+
+For example, userspace threading implementations historically used SIGALARM, where a thread registers for the kernel to deliver a signal after a certain period of time.
+Not switching that state in fibers is problematic in the presence of third-party code.
+If one function calls [`alarm`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/alarm.html) and then calls a yielding function and the resumed fiber masks SIGALARM, the alarm will not be delivered.
+Existing software may do both of these things, yet not document that they do because it's an implementation detail.
+Both of these are POSIX APIs that are not specified by C++ and so ISO C++ cannot mandate their behaviour.
+The same applies to Win32 asynchronous callbacks.
+
+The  IEEE  Std  1003.1-2004  ("POSIX.1”) revision marked the functions makecontext() and swapcontext() (the ucontext APIs) as obsolete, noting that they were intended for use for safely returning from a signal handler into the signalled frame (which is not possible with setjmp / longjmp) and were not intended to be used for threading.
+The IEEE Std 1003.1-2008 ("POSIX.1") revision removed the functions from the specification.
+
+Correctly implementing this would require code that is specific to both architectures and operating systems.
+This would make it unique in the C++ standard library.
+Some features are architecture-specific but operating-system agnostic, some are operating-system specific but architecture-agnostic.
+No existing features have this property.
+
+### Users of TLS expect it to be private
+
+Thread-local state is expected to be modified only in the same lexical scopes.
+P0876 explicitly notes this in one context: exceptions.
+The exception implementation typically maintains a list of in-flight exceptions in thread-local storage.
```

Comment:
I think page 28 says "it is forbidden to resume a fiber on any thread other than the one on which it was first resumed."

Reply:
Yes, that's fine.  It's resuming another fibre that touches the same TLS that's a problem.